### DISCUSSION TITLE
Com and Lpt ports?

### DISCUSSION THREAD

**Discussion by bob-g4bby** [fdivitto/FabGL#343] [General]
I've been experimenting with the PC Emulator. A couple of utilities report that the PC has COM1, COM2 and a printer port. What hardware is needed to provide the actual I/O pins, please? I've taken several looks inside the firmware, but don't see any pins being assigned (and the ESP32 has too few spare pins left anyway).
Cheers, Bob SW U.K.
P.S. I've enjoyed revisiting Turbo Pascal which runs really well on this tiny machine.

**Comment by lgpf** [comment]
Maybe at least a COM1 using those extra GPIO pions would be useful to connect an external modem or serial printer, There are 6 pins left exposed that we could use as COM1.

**Comment by bob-g4bby** [comment]
If two pins only are used per COM port, that's potentially 3 COM ports?

**Reply by lgpf** [reply]
Probably not. We are talking here about a full serial port to communicate with an external modem. Maybe even 6 pins wouldnt be enough. I think it requires at least 4 for data and control. But full serial cable need 6 or 7 .. not sure if all would be used by external modem.
